mediapipe/tasks/ios/text/text_classifier/sources/MPPTextClassifier.h

```diff
@@ -0,0 +1,102 @@
+ *    representing the input ids, mask ids, and segment ids. This input signature requires a
+ *    Bert Tokenizer process unit in the model metadata.
+ *  - Or one input tensor `kTfLiteInt32` of shape `[batch_size xmax_seq_len]` representing
+ *    the input ids. This input signature requires a Regex Tokenizer process unit in the
+ *    model metadata.
+ *  - Or one input tensor ({@code kTfLiteString}) that is shapeless or has shape `[1]` containing
+ * the input string.
+ *
+ * At least one output tensor `(kTfLiteFloat32}/kBool)` with:
+ *  - `N` classes and shape `[1 x N]`
+ *  - optional (but recommended) label map(s) as AssociatedFile-s with type TENSOR_AXIS_LABELS,
+ *    containing one label per line. The first such AssociatedFile (if any) is used to fill the
+ *    `class_name` field of the results. The `display_name` field is filled from the AssociatedFile
+ *    (if any) whose locale matches the `display_names_locale` field of the
+ * `MPPTextClassifierOptions` used at creation time ("en" by default, i.e. English). If none of
+ * these are available, only the `index` field of the results will be filled.
+ *
+ * @brief Performs classification on text.
+ */
+NS_SWIFT_NAME(TextClassifier)
+@interface MPPTextClassifier : NSObject
+
+/**
+ * Creates a new instance of `MPPTextClassifier` from an absolute path to a TensorFlow Lite model
+ * file stored locally on the device and the default `MPPTextClassifierOptions`.
+ *
+ * @param modelPath An absolute path to a TensorFlow Lite model file stored locally on the device.
+ *
+ * @param error An optional error parameter populated when there is an error in initializing
+ * the text classifier.
+ *
+ * @return A new instance of `MPPTextClassifier` with the given model path. `nil` if there is an
+ * error in initializing the text classifier.
+ */
+- (instancetype)initWithModelPath:(NSString *)modelPath error:(NSError **)error;
+
+/**
+ * Creates a new instance of `MPPTextClassifier` from the given `MPPTextClassifierOptions`.
+ *
+ * @param options The options of type `MPPTextClassifierOptions` to use for configuring the
+ * `MPPTextClassifier`.
+ *
+ * @param error An optional error parameter populated when there is an error in initializing
+ * the text classifier.
+ *
+ * @return A new instance of `MPPTextClassifier` with the given options. `nil` if there is an error
+ * in initializing the text classifier.
+ */
+- (instancetype)initWithOptions:(MPPTextClassifierOptions *)options error:(NSError **)error;
+
+/**
+ * Performs classification on the input text.
+ *
+ * @param text The `NSString` on which classification is to be performed.
+ *
+ * @param error An optional error parameter populated when there is an error in performing
+ * classification on the input text.
+ *
+ * @return  A `MPPTextClassifierResult` object that contains a list of text classifications.
+ */
+- (nullable MPPTextClassifierResult *)classifyWithText:(NSString *)text error:(NSError **)error;
```

Comment:
No strong opinion here, but should we drop the "With"?

Reply:
Updated this. FYI, we settled for "With" based on a review comment in the tflite support library. I prefer dropping the "With".